include/aws/common/encoding.h

```diff
@@ -156,6 +164,16 @@ AWS_STATIC_IMPL bool aws_text_is_utf8(const uint8_t *bytes, size_t size);
  */
 AWS_COMMON_API bool aws_text_is_valid_utf8(struct aws_byte_cursor bytes);
 
+/**
+ * Scans every byte, and returns true if it is valid UTF8/ASCII as defined in RFC-3629 and
+ * user's customized validation callback. The function will return true for empty string
+ * regardless of the callback functions.
+ * The text does not need to begin with a UTF8 BOM.
+ */
+AWS_COMMON_API bool aws_text_is_valid_utf8_with_callback(
+    struct aws_byte_cursor bytes,
+    aws_utf8_customzied_validator_callback_fn callback);
```

Comment:
I'd call this parameter validator maybe

Also, while it doesn't actually change anything underneath (functions are passed as pointers) I'd explicitly mark the function as a pointer.

Reply:
"validator" is already used for the utf8 validator in the function body. I will use validator_fn instead.